pkg/schema/definitions/handler_test.go

```diff
@@ -4,113 +4,131 @@ import (
-		"management.cattle.io.newresource": "io.cattle.management.v2.NewResource",
-		"noversion.cattle.io.resource":     "io.cattle.noversion.v1.Resource",
-		"missinggroup.cattle.io.resource":  "io.cattle.missinggroup.v1.Resource",
-	}
+func TestSchemaIDToCRDName(t *testing.T) {
 	tests := []struct {
-		name              string
-		openapiError      error
-		serverGroupsErr   error
-		useBadOpenApiDoc  bool
-		nilGroups         bool
-		wantModels        *proto.Models
-		wantSchemaToModel map[string]string
-		wantError         bool
+		schemaID        string
+		expectedCRDName string
 	}{
 		{
-			name:              "success",
-			wantModels:        &defaultModels,
-			wantSchemaToModel: defaultSchemaToModel,
-		},
-		{
-			name:         "error - openapi doc unavailable",
-			openapiError: fmt.Errorf("server unavailable"),
-			wantError:    true,
-		},
-		{
-			name:             "error - unable to parse openapi doc",
-			useBadOpenApiDoc: true,
-			wantError:        true,
-		},
-		{
-			name:            "error - unable to retrieve groups and resources",
-			serverGroupsErr: fmt.Errorf("server not available"),
-			wantError:       true,
+			schemaID:        "management.cattle.io.userattributes",
+			expectedCRDName: "userattributes.management.cattle.io",
 		},
 		{
-			name:       "no groups or error from server",
-			nilGroups:  true,
-			wantModels: &defaultModels,
-			wantSchemaToModel: map[string]string{
-				"management.cattle.io.globalrole":  "io.cattle.management.v1.GlobalRole",
-				"management.cattle.io.newresource": "io.cattle.management.v2.NewResource",
-				"noversion.cattle.io.resource":     "io.cattle.noversion.v1.Resource",
-				"missinggroup.cattle.io.resource":  "io.cattle.missinggroup.v1.Resource",
-			},
+			schemaID:        "pods",
+			expectedCRDName: "",
 		},
 	}
 	for _, test := range tests {
 		test := test
-		t.Run(test.name, func(t *testing.T) {
-			t.Parallel()
-			client, err := buildDefaultDiscovery()
-			client.DocumentErr = test.openapiError
-			client.GroupsErr = test.serverGroupsErr
-			if test.useBadOpenApiDoc {
-				schema := client.Document.Definitions.AdditionalProperties[0]
-				schema.Value.Type = &openapi_v2.TypeItem{
-					Value: []string{"multiple", "entries"},
-				}
-			}
-			if test.nilGroups {
-				client.Groups = nil
-			}
-			require.Nil(t, err)
-			handler := SchemaDefinitionHandler{
-				client: client,
-			}
-			err = handler.Refresh()
-			if test.wantError {
-				require.Error(t, err)
-			} else {
-				require.NoError(t, err)
-			}
-			require.Equal(t, test.wantModels, handler.models)
-			require.Equal(t, test.wantSchemaToModel, handler.schemaToModel)
+		t.Run(test.schemaID, func(t *testing.T) {
+			crdName := schemaIDToCRDName(test.schemaID)
+			require.Equal(t, test.expectedCRDName, crdName)
 		})
-
 	}
 }
 
-func Test_byID(t *testing.T) {
-	defaultDocument, err := openapi_v2.ParseDocument([]byte(openapi_raw))
+func TestRefresh(t *testing.T) {
```

Comment:
The previous test code was using private fields of the handler to change the behavior. This made it difficult for me when I was working on this since initially I changed the implementation quite a bit. (eg: the models/schemaToModel fields were no longer in the handler).

Those fields are now back in the handler so I could revert the change here, but I felt this way also worked fine. Let me know what you think.

Reply:
Agreed, once I have a design I'm okay with code wise I'll revisit this. I'll probably be reverting the changes to the test.